python/lsst/daf/butler/dimensions/_group.py

```diff
@@ -449,6 +450,24 @@ def add_to_order(element: DimensionElement) -> None:
         order.extend(element for element in self.elements if element not in done)
         return tuple(order)
 
+    @property
+    def region_dim(self) -> TopologicalRelationshipEndpoint | None:
+        """Return the most appropriate spatial dimension to use when looking
+        up a region.
+        """
+        for family in self.spatial:
+            return family.choose(self.elements, self.universe)
```

Comment:
This is almost never going to matter because most dimension groups of interest have only one spatial or temporal family, but I'm a little uncomfortable just returning the first one when there is more than one.  The relevant real-world case is warps, which have `{visit, patch}` dimensions, and I don't know which this would pick in this case (I'm not even positive it will be deterministic; it might be set-iteration-order-dependent).

There is analogous logic in the `DataCoordinate.region` property that computes the intersection of the regions for the two families, which I think makes sense for warps but might not for more unusual cases, and it's not really an option here (returning a sequence of names whose regions need to be intersected later acts against making it easy to use).  I think it might make it best to just have this return `None` if there is more than one family.